src/sim_engine/pool.rs

```diff
@@ -104,6 +114,19 @@ impl Pool for SimPool {
         Ok(())
     }
 
+    fn create_snapshot(&mut self, snapshot_name: &str, source: &str) -> EngineResult<()> {
+
+        let parent_id = try!(self.check_snapshot_source(source));
+
+        try!(self.create_filesystem(&snapshot_name, &String::from(""), None));
```

Comment:
What would happen if you changed the signature of ```create_filesystem``` so that it returned the just created filesystem object?

Maybe it would be helpful for ```create_snapshot``` to do something similar.